api/notd/collection_processor.py

```diff
@@ -0,0 +1,51 @@
+import asyncio
+import logging
+
+from core.exceptions import NotFoundException
+from core.requester import Requester
+from core.requester import ResponseException
+
+from notd.model import RetrievedCollection
+
+
+class CollectionDoesNotExist(NotFoundException):
+    pass
+
+class CollectionProcessor:
+    def __init__(self, requester: Requester):
+        self.requester = requester
+
+    async def retrieve_collection(self,address) -> RetrievedCollection:
+        response = 0
+        retryCount = 0
```

Comment:
you need to get the name and symbol from the contract too.

Reply:
Do you mean as input arguments?

Reply:
no i mean you should query the blockchain at this address to get the name and address as part of the ERC721 standard